core/src/main/kotlin/com/malinskiy/marathon/execution/device/DeviceActor.kt

```diff
@@ -187,21 +211,19 @@ class DeviceActor(private val devicePoolId: DevicePoolId,
         job = async {
             try {
                 device.execute(configuration, devicePoolId, batch, result, progressReporter)
-            } catch (e: DeviceLostException) {
-                logger.error(e) { "Critical error during execution" }
+            } catch (exc: TestBatchTimeoutException) {
+                logger.warn { "Critical error during batch execution: batch timed out. ${exc.cause.toString()}" }
+                state.transition(DeviceEvent.Terminate)
```

Comment:
this event suggests that device will no longer be available. is that the case?

Reply:
Actually not - but that was the default behavior for such a case. I intend to later experiment with moving the device into the Initialize state.